src/Validation/SavePostValidationEvent.php

```diff
@@ -0,0 +1,139 @@
+<?php
+/**
+ * Single cron event to validate a saved post's permalink in the background.
+ *
+ * @package AMP
+ * @since 2.1
+ */
+
+namespace AmpProject\AmpWP\Validation;
+
+use AmpProject\AmpWP\BackgroundTask\BackgroundTaskDeactivator;
+use AmpProject\AmpWP\BackgroundTask\SingleScheduledBackgroundTask;
+use AmpProject\AmpWP\DevTools\UserAccess;
+
+/**
+ * SavePostValidationEvent class.
+ *
+ * @since 2.1
+ *
+ * @internal
+ */
+final class SavePostValidationEvent extends SingleScheduledBackgroundTask {
+
+	/**
+	 * Instance of URLValidationProvider
+	 *
+	 * @var URLValidationProvider
+	 */
+	private $url_validation_provider;
+
+	/**
+	 * Instance of UserAccess.
+	 *
+	 * @var UserAccess
+	 */
+	private $dev_tools_user_access;
+
+	/**
+	 * The cron action name.
+	 *
+	 * @var string
+	 */
+	const BACKGROUND_TASK_NAME = 'amp_single_post_validate';
+
+	/**
+	 * Class constructor.
+	 *
+	 * @param BackgroundTaskDeactivator $background_task_deactivator Background task deactivator instance.
+	 * @param UserAccess                $dev_tools_user_access Dev tools user access class instance.
+	 */
+	public function __construct( BackgroundTaskDeactivator $background_task_deactivator, UserAccess $dev_tools_user_access ) {
+		parent::__construct( $background_task_deactivator );
+
+		$this->dev_tools_user_access = $dev_tools_user_access;
+	}
+
+	/**
+	 * Callback for the cron action.
+	 *
+	 * @param mixed ...$args The args received with the action hook where the event was scheduled.
+	 */
+	public function process( ...$args ) {
+		$post_id = reset( $args );
+
+		if ( empty( get_post( $post_id ) ) ) {
```

Comment:
Since `$post_id` being `0` can result in `get_post()` returning the global `$post`.

```suggestion
		if ( empty( $post_id ) || empty( get_post( $post_id ) ) ) {
```